### PR TITLE
Updated development dependencies

### PR DESCRIPTION
**Pull request by ijlee2**
## Description

Getting the project ready for possible Hacktoberfest issues.

Because Dependabot doesn't know why Ember-related packages _don't need to_ exist, there were a few installed packages that can now be removed or replaced with another package (e.g. `ember-ajax`, `ember-cli-template-lint`, `ember-cli-eslint`, `ember-cli-uglify`).

## How to Review

I recommend reviewing commits one by one.

## Notes

1\. After updating `eslint` and `ember-template-lint`, I encountered several lint violations in the app. For now, I disallowed lint checking in the problematic files. I plan to open issues to fix the problem for Hacktoberfest.

2\. After updating `ember-data` (jumped 3 minor versions), you may encounter the following error message when you run `ember s`:

```bash
Build Error (broccoli-persistent-filter:Babel > [Babel: @ember-data/store]) in -private/system/references/reference.ts
9:54:25 AM: /opt/build/repo/-private/system/references/reference.ts: Imported DEPRECATE_REFERENCE_INTERNAL_MODEL from @ember-data/private-build-infra/deprecations which is not a supported flag.
```

Locally, you can fix this issue by deleting `node_modules` and running `npm install`. For Netlify, which employs a cache for build, you will want to make sure to click `Retry deploy` button, then select `Clear cache and deploy site` option.

Issue reported and fixed in: https://github.com/emberjs/data/issues/7221

3\. There were a couple of packages that I didn't feel comfortable updating: `algoliasearch` and `ember-styleguide`. The latter introduced a [major release (v5.0.0)](https://github.com/ember-learn/ember-styleguide/releases/tag/v5.0.0) with a breaking change. If I understood the release notes correctly, this app will need to update its stylesheets for compatibility. I'm not sure yet if the scope of the work is small enough for Hacktoberfest, or the work can be better done by the website strike team.

### CODE REVIEW

File: .template-lintrc.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+module.exports = {
+  extends: 'octane',
+
+  // TODO: Turn on a rule and fix errors
+  rules: {
```
Comment (ijlee2) [inline .template-lintrc.js:7]: These are exceptions that I had to make after installing the latest `ember-template-lint`. We'll open issues to Octanify the app and remove these exceptions.

File: lib/content-docs-generator/index.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 'use strict';
 
+// eslint-disable-next-line node/no-extraneous-require
```
Comment (ijlee2) [inline lib/content-docs-generator/index.js:3]: Temporary exceptions.

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 language: node_js
 node_js:
-  - "10"
+  - "12"
```
Comment (ijlee2) [inline .travis.yml:4]: This better matches the build environment for Netlify (it uses `v12.18.1`).

**Comment by jaredgalanis** [comment]
@ijlee2 do you expect that we'll have to press the retry deploy option on Netlify for the time being or at least until things are fixed in Ember Data (or wherever the problem is coming from)? I think that's fine for now, but I think it may not be an acceptable outcome if it persists for a long while.

**Comment by ijlee2** [comment]
@jaredgalanis I'm hoping it's only once, if merging this PR to `master` resulted in a failure.

When I merged https://github.com/ember-learn/upgrade-guide/pull/51 (similarly had to update Ember Data to 3.21), I think I had to clear cache and retry the build. Subsequent PRs didn't cause any issues.

**Comment by jaredgalanis** [comment]
Yeah, I had to do it for two dependabot PRs that were opened since you merged your work which makes me a little worried that it's going to be an ongoing issue. Let's monitor and see.

**Pull request merged**